### DISCUSSION TITLE
[Q]call of overloaded 'getParam(int)' is ambiguous

### DISCUSSION THREAD

**Discussion by maxelle** [mathieucarbou/ESPAsyncWebServer#67] [Q&A]
I receive this error compiling

C:\Arduino2\test\test.ino:647:47: error: call of overloaded 'getParam(int)' is ambiguous
  647 |       AsyncWebParameter* p = request->getParam(0);

Please help
Thanks Max

**Comment by Pablo2048** [comment]
See https://github.com/mathieucarbou/ESPAsyncWebServer/releases/tag/v3.0.1 - you have to add `const` keyword.
```cpp
const AsyncWebParameter *p = request->getParam(0);
```

**Comment by maxelle** [comment]
Thanks for answer,
I tryed bt same error... 
![image](https://github.com/user-attachments/assets/5c8d0a12-3085-40a1-9e7f-d5abda8f9b29)

**Comment by mathieucarbou** [comment]
@maxelle : your parameter type is wrong. Please look at the header file. This is the doc.

**Comment by maxelle** [comment]
so i have to cast like...
const AsyncWebParameter* p = request->getParam((size_t)0);

**Comment by mathieucarbou** [comment]
> so i have to cast like... const AsyncWebParameter* p = request->getParam((size_t)0);

This is one way to do it, yes. There are other ways also.
`0` as written here is an `int`. There is no `getParam()` accepting `int`. And the compiler cannot know which one to pick.
